src/config.c

```diff
@@ -3039,6 +3039,7 @@ standardConfig static_configs[] = {
     createBoolConfig("lazyfree-lazy-user-flush", NULL, DEBUG_CONFIG | MODIFIABLE_CONFIG, server.lazyfree_lazy_user_flush, 0, NULL, NULL),
     createBoolConfig("repl-disable-tcp-nodelay", NULL, MODIFIABLE_CONFIG, server.repl_disable_tcp_nodelay, 0, NULL, NULL),
     createBoolConfig("repl-diskless-sync", NULL, DEBUG_CONFIG | MODIFIABLE_CONFIG, server.repl_diskless_sync, 1, NULL, NULL),
+    createBoolConfig("repl-rdb-connection", NULL, DEBUG_CONFIG | MODIFIABLE_CONFIG | HIDDEN_CONFIG, server.rdb_conn_enabled, 0, NULL, NULL),
```

Comment:
[Readability suggestion]
1. add the `-enabled` suffix to align with existing configs like `cluster-enabled`
2. add `parallel` and `fullsync` to emphasize the key idea of this change
3. I would suggest starting with IMMUTABLE first so we don't have to worry about transient states where the config is toggled while a full sync is underway
4. Can we consider removing `rdb` because this design is actually file format agnostic and we have an ongoing discussion #59?
```suggestion
    createBoolConfig("repl-parallel-fullsync-enabled", NULL, DEBUG_CONFIG | IMMUTABLE_CONFIG | HIDDEN_CONFIG, server.rdb_conn_enabled, 0, NULL, NULL),
```

Reply:
There is no real parallel fullsync here, not sure its a good name. What do you think of `dual-connection-sync-enabled`?
Previously, we decided to retain this configuration option solely for debugging. We can revisit that decision now, as merging the feature with the flag disabled will likely be smoother. If we do that, we should probably make this immutable as you suggested. 
WDYT?

Reply:
>There is no real parallel fullsync here, not sure its a good name.

I was thinking that the (normal) replication and the full sync are running in parallel but maybe "parallel fullsync" is confusing - as if we are saying the fullsync itself can be broken into parallel parts. 

"dual-connection-sync-enabled" sounds good to me (should we add "full" there?)

>Previously, we decided to retain this configuration option solely for debugging. We can revisit that decision now,

I think the last discussion (with @madolson and @ranshid) was more on "stability", i.e., what is the best/safest way to introduce a major feature like this one. I can see both sides of the argument: stability vs "discoverability" (meaning that keeping it off by default might lead to no/little usage). I would say let's keep this IMMUTABLE config off by default for now. After merging this PR, we should find ways to increase the test coverage. We can easily flip the default state late in the release cycle, if we feel conformable.

Reply:
+1 for "dual-connection-sync-enabled" (doesn't need "full")

If it's not more complex, then mutable is better, isn't it? If we only check the config when we send REPLCONF CAPA and not when we receive +FULLSYNCNEEDED, then we're fine, I guess. (A test case for toggling the config in the middle of this won't hurt though.)

Reply:
`dual-connection-sync-enabled` it is:)

IMHO if we find a bug in the future, mutable allows us to turn it off without replacing the server, which is a nice option to have. Also using mutable makes testing easier. I can add tests for different on/off situations. 
Anyway we can mark it as `PROTECTED_CONFIG` to prevent accidental changes.